app/services/developer_service.rb

```diff
@@ -6,9 +6,21 @@ def initialize(params)
 
   def register_as_developer
     if @user && !@user.developer?
-      ApiKey.create(user: @user)
-      @user.roles << Role.find_or_create_by(name: 'developer')
+      key = generate_key
+      @user.create_api_key(key: key)
+      raise ArguementError if !@user.api_key
+      make_developer_role(@user)
     end
   end
 
+  def generate_key
+    Digest::SHA512.hexdigest(Time.now.to_s + @user.email)
+  end
+
+  private
+
+    def make_developer_role(user)
+      user.roles << Role.find_or_create_by(name: 'developer')
```

Comment:
@liambarstad Do we need to take in the user parameter on this method? Looks like we could use the user sent in in initialize.

Reply:
Speaking of... the parameter on initialize is called `params` but you only care about the user right now. I think we should change the parameter to `user`.